src/snark/src/zk_proof_systems/ppzksnark/r1cs_ppzksnark/r1cs_ppzksnark.tcc

```diff
@@ -451,81 +521,36 @@ r1cs_ppzksnark_proof<ppT> r1cs_ppzksnark_prover(const r1cs_ppzksnark_proving_key
     const qap_witness<Fr<ppT> > qap_wit = r1cs_to_qap_witness_map(constraint_system, primary_input, auxiliary_input, d1, d2, d3);
     leave_block("Compute the polynomial H");
 
-#ifdef DEBUG
-    const Fr<ppT> t = Fr<ppT>::random_element();
-    qap_instance_evaluation<Fr<ppT> > qap_inst = r1cs_to_qap_instance_map_with_evaluation(constraint_system, t);
-    assert(qap_inst.is_satisfied(qap_wit));
-#endif
-
-    knowledge_commitment<G1<ppT>, G1<ppT> > g_A = pk.A_query[0] + qap_wit.d1*pk.A_query[qap_wit.num_variables()+1];
-    knowledge_commitment<G2<ppT>, G1<ppT> > g_B = pk.B_query[0] + qap_wit.d2*pk.B_query[qap_wit.num_variables()+1];
-    knowledge_commitment<G1<ppT>, G1<ppT> > g_C = pk.C_query[0] + qap_wit.d3*pk.C_query[qap_wit.num_variables()+1];
-
-    G1<ppT> g_H = G1<ppT>::zero();
-    G1<ppT> g_K = (pk.K_query[0] +
-                   qap_wit.d1*pk.K_query[qap_wit.num_variables()+1] +
-                   qap_wit.d2*pk.K_query[qap_wit.num_variables()+2] +
-                   qap_wit.d3*pk.K_query[qap_wit.num_variables()+3]);
-
-#ifdef DEBUG
-    for (size_t i = 0; i < qap_wit.num_inputs() + 1; ++i)
-    {
-        assert(pk.A_query[i].g == G1<ppT>::zero());
-    }
-    assert(pk.A_query.domain_size() == qap_wit.num_variables()+2);
-    assert(pk.B_query.domain_size() == qap_wit.num_variables()+2);
-    assert(pk.C_query.domain_size() == qap_wit.num_variables()+2);
-    assert(pk.H_query.size() == qap_wit.degree()+1);
-    assert(pk.K_query.size() == qap_wit.num_variables()+4);
-#endif
```

Comment:
These debug assertions have been dropped. Is this intentional?

Reply:
@ebfull ?

Reply:
They could be left in.

Reply:
I think this can't be left in cause it needs all the proving key at once.

Reply:
Most of this could be split across the various functions. The for loop would be harder though, and given that we don't set `DEBUG`, I'm fine with leaving these out.